common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/OAuth2Strategy.java

```diff
@@ -260,10 +269,26 @@ protected final void setIssuer(final Uri issuer) {
      *
      * @param response Http response.
      */
-    protected abstract GenericTokenResult getTokenResultFromHttpResponse(HttpResponse response);
+    protected abstract GenericTokenResult getTokenResultFromHttpResponse(HttpResponse response) throws ClientException;
 
     // TODO
 //    protected abstract void validateAuthorizationResponse(GenericAuthorizationResponse response);
 
-//    protected abstract void validateTokenResponse(GenericTokenResponse response);
+    protected abstract void validateTokenResponse(GenericTokenResponse response) throws ClientException;
+
+    /**
+     * Validate the result yielded from the cache prior to returning it to the caller.
+     *
+     * @param cacheRecord The {@link ICacheRecord} to validate.
+     * @return True if the CacheRecord checks out. False otherwise.
+     */
+    public boolean validateCachedResult(@NonNull final ICacheRecord cacheRecord) {
+        // TODO Perform validations on the CacheRecord prior to returning result...
+
+        // TODO Ideally, we would check the expiry, completeness of record, etc here...
+        // TODO This looks like there's a bit of refactoring involved to account for broker cases...
```

Comment:
Improve the comment here -- clarify that this requires refactoring of the controllers

Reply:
Resolved in a74c7d8c624ab839a9f61b89f63963671b41e9a9